### PR TITLE
Upgrade test apps to react native 0.59

### PR DESCRIPTION
**Pull request by dtrabo**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Upgraded TestApp, TestAppTypeScript, DemoApp to RN 0.59.0

### CODE REVIEW

File: DemoApp/package.json [hunk 1]
```diff
@@ -13,34 +13,36 @@
     "appcenter-crashes": "^1.12.2",
     "appcenter-push": "^1.12.2",
     "metro": "0.48.1",
-    "react": "16.6.3",
-    "react-native": "0.58.3",
+    "react": "16.8.3",
+    "react-native": "^0.59.1",
```
Comment (dhei) [inline DemoApp/package.json]: use exact version instead of `^` in package.json

File: DemoApp/ios/DemoApp/AppDelegate.h [hunk 2]
```diff
@@ -7,9 +7,11 @@
  * of patent rights can be found in the PATENTS file in the same directory.
  */
 
+#import <React/RCTBridgeDelegate.h>
 #import <UIKit/UIKit.h>
 
 @interface AppDelegate : UIResponder <UIApplicationDelegate>
+@interface AppDelegate : UIResponder <UIApplicationDelegate, RCTBridgeDelegate>
```
Comment (dhei) [inline DemoApp/ios/DemoApp/AppDelegate.h]: Duplicated `AppDelegate` declarations.

Comment (dhei) [inline DemoApp/ios/DemoApp/AppDelegate.h]: Why importing `RCTBridgeDelegate` even though we're not using it? Is it added automatically by `react-native upgrade` or some script?

Comment (dtrabo) [reply DemoApp/ios/DemoApp/AppDelegate.h]: I leaned on these changes https://github.com/react-native-community/rn-diff-purge/compare/version/0.58.6...version/0.59.0. I'm not sure, but It seems to be necessary

Comment (MatkovIvan) [reply DemoApp/ios/DemoApp/AppDelegate.h]: Please do the same for other apps

Comment (dhei) [reply DemoApp/ios/DemoApp/AppDelegate.h]: @dtrabo okay, that's already.

**Comment by dhei** [comment]
There is an issue of `react-navigation` with jest framework blocking the upgrade to RN 0.59.x. The error stacktrace from jest looks like this:
```
TypeError: Cannot read property 'default' of undefined
```
and full stacktrace can be found at [here](https://gist.github.com/annakocheshkova/03160096fe15623892fe7ffbae72c969).

We reported https://github.com/react-navigation/react-navigation/issues/5742 on `react-navigation` to track this.

**Comment by annakocheshkova** [comment]
Applied several fixes from [this thread](https://github.com/facebook/react-native/issues/22175).

**Pull request merged**